### PR TITLE
show purchases

### PR DESCRIPTION
**Pull request by sguan12**
Transactions adapter for recycler view, populated recycler view from backend, attached detail fragment on click, formatted date and display ui a bit

### CODE REVIEW

**Comment by sguan12** [comment]
image

image

File: app/src/main/java/com/example/coinage/fragments/AddFragment.java [hunk 1]
```diff
@@ -31,9 +32,9 @@
 public class AddFragment extends Fragment {
     public static final String TAG = "AddFragment";
 
-    public static final Calendar myCalendar = Calendar.getInstance();
+    public final Calendar myCalendar = Calendar.getInstance();
```
Comment (chetfb) [inline app/src/main/java/com/example/coinage/fragments/AddFragment.java:35]: Nice!

More details on static variables and static keyword may be found here:
https://www.edureka.co/blog/static-keyword-in-java/#StaticVariable

Just food for thought:
Static keyword essentially creates a globally unique instances -- this may lead to issues later if data needs to be properly encapsulated -- and may be even buggy if the instance is written+read from multiple threads ...
so you need to use them only when you truly need a single global instance especially like constant strings and constant numbers which are globally unique readonly objects.

**Review by chetfb — APPROVED** [review]

**Comment by chetfb** [comment]
The code is coming along very beautifully ... so may features are getting implemented! Nice job!

File: app/src/main/java/com/example/coinage/fragments/AddFragment.java (hunk 1 above)
Comment (sguan12) [reply app/src/main/java/com/example/coinage/fragments/AddFragment.java:35]: Sounds good, I'll keep that in mind!

File: app/src/main/java/com/example/coinage/LoginActivity.java [hunk 2]
```diff
@@ -48,29 +54,34 @@ public void onClick(View v) {
         });
     }
 
-    private void goHomeActivity() {
+    private void goMainActivity() {
+        Log.i(TAG, "go main activity");
```
Comment (wgregmeta) [inline app/src/main/java/com/example/coinage/LoginActivity.java:58]: It's ok  to log full sentences that ideally could make sense to someone who saw them without knowing the code yet. Something like "Navigating to main activity..." could be a little better

**Comment by wgregmeta** [comment]
Agree with @chetfb, great pace!!

**Comment by wgregmeta** [comment]
Also I just commented on an old PR but quicktime player is a great way to take videos to show the feature working :)

File: app/src/main/java/com/example/coinage/fragments/DetailFragment.java [hunk 3]
```diff
@@ -1,20 +1,35 @@
 package com.example.coinage.fragments;
```
Comment (wgregmeta) [inline app/src/main/java/com/example/coinage/fragments/DetailFragment.java:1]: nit - Could be a tiny bit more descriptive with these file names, so someone could check out your code for the first time and have the best chance of knowing what a file does (maybe `TransactionDetailFragment` or something)

File: app/src/main/java/com/example/coinage/RegisterActivity.java [hunk 4]
```diff
@@ -16,6 +16,7 @@
 
 public class RegisterActivity extends AppCompatActivity {
     public static final String TAG = "RegisterActivity";
+
```
Comment (wgregmeta) [inline app/src/main/java/com/example/coinage/RegisterActivity.java:19]: extra whitespace should go in separate PR
